bookstore/client/nb_client.py

```diff
@@ -1,94 +1,41 @@
-#   'port': 8888,
-#   'secure': False,
-#   'token': '',
-#   'url': 'http://localhost:8888/'}]
-# ```
-
+"""Client to test bookstore endpoints from within a notebook.
+
+
+TODO: (Clarify) We want to test our bookstore endpoints, but it's no fun having
+to do this in an insecure fashion. Better would be to have some security in
+place.
+
+Example
+-------
+
+[{'base_url': '/',
+  'hostname': 'localhost',
+  'notebook_dir': '/Users/mpacer/jupyter/eg_notebooks',
+  'password': False,
+  'pid': 96033,
+  'port': 8888,
+  'secure': False,
+  'token': '',
+  'url': 'http://localhost:8888/'}]
+"""
 import os
-
-import requests
-
 from copy import deepcopy
 from typing import NamedTuple
 
-
+import requests
+from IPython import get_ipython
 from notebook.notebookapp import list_running_servers
 
 
-def extract_kernel_id(connection_file):
-    return os.path.basename(connection_file).lstrip('kernel-').rstrip('.json')
-
-
-class LiveNotebookRecord(NamedTuple):
-    """Representation of live notebook server
-
-    This is a realization of the object returned by
-    `notebook.notebookapp.list_running_servers()`.
-    """
-
-    base_url: str
-    hostname: str
-    notebook_dir: str
-    password: bool
-    pid: int
-    port: int
-    secure: bool
-    token: str
-    url: str
-
-
-class KernelInfo:
-    #     id: str # 'f92b7c8b-0858-4d10-903c-b0631540fb36',
-    #     name: str # 'dev',
-    #     last_activity: str #'2019-03-14T23:38:08.137987Z',
-    #     execution_state: str #'idle',
-    #     connections: int # 0
-    def __init__(self, *args, id, name, last_activity, execution_state, connections):
-        self.id = id
-        self.name = name
-        self.last_activity = last_activity
-        self.execution_state = execution_state
-        self.connections = connections
-
-
-class NotebookSession:  # (NamedTuple):
-    #     id: str #'68d9c58f-c57d-4133-8b41-5ec2731b268d',
-    #     path: str #'Untitled38.ipynb',
-    #     name: str #'',
-    #     type: str #'notebook',
-    #     kernel: KernelInfo
-    #     notebook: dict # {'path': 'Untitled38.ipynb', 'name': ''}}}
-
-    def __init__(self, *args, path, name, type, kernel, notebook, **kwargs):
-        self.path = path
-        self.name = name
-        self.type = type
-        self.kernel = KernelInfo(**kernel)
-        self.notebook = notebook
-
-
 class NotebookClient:
+    """Client used to interact with bookstore from within a running notebook UI"""
+
     def __init__(self, nb_config):
         self.nb_config = nb_config
-        self.nb_record = LiveNotebookRecord(**self.nb_config)
-        self.url = self.nb_record.url.rstrip(
-            "/"
-        )  # So that we can have full API endpoints without double //
-        self.token = self.nb_record.token
+        self.nbserver_record = LiveNotebookServerRecord(**self.nb_config)
+        # Right strip '/' so we can have full API endpoints without double '//'
+        self.url = self.nbserver_record.url.rstrip("/")
```

Comment:
we might be able to reuse a notebook utility for this functionality. I'll look into it.